cmd/create/operatorroles/cmd.go

```diff
@@ -0,0 +1,379 @@
+	)
+
+	flags.StringVar(
+		&args.mode,
+		"mode",
+		modes[0],
+		"How to perform the operation. Valid options are:\n"+
+			"auto: Roles will be created using the current AWS account\n"+
+			"manual: Role files will be saved in the current directory",
+	)
+	Cmd.RegisterFlagCompletionFunc("mode", modeCompletion)
+
+	confirm.AddFlag(flags)
+	interactive.AddFlag(flags)
+}
+
+func modeCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return modes, cobra.ShellCompDirectiveDefault
+}
+
+func run(cmd *cobra.Command, _ []string) {
+	reporter := rprtr.CreateReporterOrExit()
+	logger := logging.CreateLoggerOrExit(reporter)
+
+	// Check that the cluster key (name, identifier or external identifier) given by the user
+	// is reasonably safe so that there is no risk of SQL injection:
+	clusterKey := args.clusterKey
+	if !ocm.IsValidClusterKey(clusterKey) {
+		reporter.Errorf(
+			"Cluster name, identifier or external identifier '%s' isn't valid: it "+
+				"must contain only letters, digits, dashes and underscores",
+			clusterKey,
+		)
+		os.Exit(1)
+	}
+
+	// Determine if interactive mode is needed
+	if !interactive.Enabled() && !cmd.Flags().Changed("mode") {
+		interactive.Enable()
+	}
+
+	// Create the AWS client:
+	awsClient, err := aws.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create AWS client: %v", err)
+		os.Exit(1)
+	}
+
+	creator, err := awsClient.GetCreator()
+	if err != nil {
+		reporter.Errorf("Failed to get IAM credentials: %s", err)
+		os.Exit(1)
+	}
+
+	// Create the client for the OCM API:
+	ocmClient, err := ocm.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create OCM connection: %v", err)
+		os.Exit(1)
+	}
+	defer func() {
+		err = ocmClient.Close()
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	// Try to find the cluster:
+	reporter.Debugf("Loading cluster '%s'", clusterKey)
+	cluster, err := ocmClient.GetCluster(clusterKey, creator)
+	if err != nil {
+		reporter.Errorf("Failed to get cluster '%s': %v", clusterKey, err)
+		os.Exit(1)
+	}
+
+	if cluster.State() != cmv1.ClusterStatePending {
+		reporter.Errorf("Cluster '%s' is %s and does not need additional configuration.",
+			clusterKey, cluster.State())
+		os.Exit(1)
+	}
+
+	if cluster.AWS().STS().RoleARN() == "" {
+		reporter.Errorf("Cluster '%s' is not an STS cluster.", clusterKey)
+		os.Exit(1)
+	}
+
+	prefix := args.prefix
+	if interactive.Enabled() {
+		prefix, err = interactive.GetString(interactive.Input{
+			Question: "Role prefix",
+			Help:     cmd.Flags().Lookup("prefix").Usage,
+			Default:  prefix,
+			Required: true,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid role prefix: %s", err)
+			os.Exit(1)
+		}
+	}
+	if prefix == "" {
+		prefix = cluster.Name()
+	}
+	if len(prefix) > 32 {
```

Comment:
I'm not sure if I asked this on the last PR, is there any reason behind `32` being the length of the prefix? `64` is the ultimate limit and I assume you just picked an arbitrary limit? (I don't disagree with it just wanted to ask)

Reply:
it's arbitrary - but if we allow too long of a prefix, we run the risk of duplicate role names. For example
`thisisaverylongrolenameprefixjusttoshowwhyitsabadidea-openshif` <-- all operator roles, when truncated to 64, would end up with the same name.

Reply:
Make sense, I was curious. Thanks.